### PR TITLE
[receiver/wavefront] wrap metrics receiver under carbon receiver instead of using export function

### PR DESCRIPTION
**Pull request by sakulali**
**Description:** 
Wavefrontreceiver is very similar to carbonreceiver: it is TCP based in which each received text line represents a single metric data point. In order to avoid using exported function `carbonreceiver.New(...)`, we can wrap metrics receiver under carbon receiver.

**Link to tracking Issue:** 
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27248

**Testing:** 
make chlog-validate
go test for wavefrontreceiver

**Documentation:**

### CODE REVIEW

File: receiver/wavefrontreceiver/receiver.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		},
+	}
+
+	carbonReceiver, err := fact.CreateMetricsReceiver(ctx, r.set, carbonCfg, r.nextConsumer)
+	if err != nil {
+		return err
+	}
+	r.carbonReceiver = carbonReceiver
+
+	err = r.carbonReceiver.Start(ctx, host)
```
Comment (songy23) [inline receiver/wavefrontreceiver/receiver.go:64]: nit: can simply do `return r.carbonReceiver.Start(ctx, host)`

**Review by songy23 — APPROVED** [review]
Minor comment, otherwise LGTM

**Review by atoulme — APPROVED** [review]
LGTM @samiura wdyt?

**Comment by sakulali** [comment]
It seems this PR ready?

**Comment by atoulme** [comment]
It's ready. @samiura last call?

**Review by samiura — APPROVED** [review]

**Comment by samiura** [comment]
@atoulme any idea why I do not have the blue sign like you folks? Does that come with special privileges or status?

**Comment by atoulme** [comment]
It's because you're a codeowner, but not a triager, approver or maintainer. See https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#roles

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**